bolt-design-system/bolt
# Convert Ordered and Unordered lists components to web component

## Jira

http://vjira2:8080/browse/BDS-532

## Summary

Refactors the in-progress work converting the bolt-unordered-list and bolt-ordered-list components to be web components.

## Details

- Simplifies the names of the components
- Breaks out the list item shared by both components into a separate `<bolt-li>` element that's required by both `<bolt-ol>` and `<bolt-ul>`
- Added logic to allow the `<bolt-ul>` component to be nested multiple levels deep and be styled correctly without requiring any config options on the nested components themselves
- Reorganizes other list-like components into a new `lists` sub-folder under `packages/components`
- Handles adjustment in styles + updates the depth of the component if a parent list or list item is dragged to a different level (ie. indented or outdented)
- Adjusted Jest snapshot diff logic to help auto-format HTML being tested against -- helps prevent against snapshot test fails like I've been seeing locally

## How to test

Run this code locally. Navigate to Components -> Ordered List / Unordered List -> View All. Check if all elements are rendered properly with javascript on and off.
